### PR TITLE
Copy menu update

### PR DESCRIPTION
**Pull request by maarksman**
Pull request for copy menu bug

### CODE REVIEW

File: src/smc-webapp/projects.cjsx [hunk 1]
```diff
@@ -502,7 +502,7 @@ class ProjectsStore extends Store
                     return pay
 
     is_deleted: (project_id) =>
-        return !!@getIn(['project_map', project_id, 'deleted'])
+        return !!@getIn(['project_map', project_id, 'deleted']) #getIn returns nested data form list pointers. e.g. 'project_map' points to project_id, which points to 'deleted.
```
Comment (williamstein) [inline src/smc-webapp/projects.cjsx]: In the comment, `‘deleted` has an opening quote but no quote.  Also, I find the comment hard to read.  Finally, it makes no sense to explain what getIn is in this *one* place.   It’s used in hundred of places; it’s better to not have a comment here, and instead have a comment in some dev guide...?

File: src/smc-webapp/projects.cjsx [hunk 2]
```diff
@@ -513,7 +513,7 @@ class ProjectsStore extends Store
             return
         account_id = webapp_client.account_id
         list = []
-        if current? and map.has(current)
+        if current? and map.has(current)   #current is for current project
```
Comment (williamstein) [inline src/smc-webapp/projects.cjsx]: Our style for comments is to put a space after the hash, so `# current is for current project`.  But might be to be more descriptive, e.g., `# current is the project_id of the current project`.

File: src/smc-webapp/projects.cjsx [hunk 3]
```diff
@@ -525,7 +525,7 @@ class ProjectsStore extends Store
             return 0
         others = []
         for i in v
-            if not i.deleted and (show_hidden or not i.get('users').get(account_id).get('hide'))
+            if (not (i.get('deleted') == true )) and (show_hidden or not i.get('users').get(account_id).get('hide'))
```
Comment (williamstein) [inline src/smc-webapp/projects.cjsx]: Change `not (i.get(‘deleted’) == true)` to `not i.get(‘deleted’)`.

File: src/smc-webapp/projects.cjsx (hunk 1 above)
Comment (maarksman) [inline src/smc-webapp/projects.cjsx]: Took the comment out since it was for personal use

File: src/smc-webapp/projects.cjsx (hunk 2 above)
Comment (maarksman) [inline src/smc-webapp/projects.cjsx]: Fixed

File: src/smc-webapp/projects.cjsx (hunk 3 above)
Comment (maarksman) [inline src/smc-webapp/projects.cjsx]: Any reason for the preferred syntax? I wasn't sure what that would return since projects which aren't deleted simply don't have a 'deleted' key/property in their map.  If 'deleted' isn't true this is avoided.

File: src/smc-webapp/projects.cjsx [hunk 4]
```diff
@@ -502,7 +502,7 @@ class ProjectsStore extends Store
                     return pay
 
     is_deleted: (project_id) =>
-        return !!@getIn(['project_map', project_id, 'deleted'])
+        return !!@getIn(['project_map', project_id, 'deleted'])
```
Comment (williamstein) [inline src/smc-webapp/projects.cjsx]: This part of the diff does not but add a blank space at the end of the line.  Please enable “Strip trailing whitespace: remove whenever file is saved” in CoCalc account settings, then open the file, then save it (with cursor not in live 505).  Then make another commit.

File: src/smc-webapp/projects.cjsx [hunk 5]
```diff
@@ -525,7 +525,8 @@ class ProjectsStore extends Store
             return 0
         others = []
         for i in v
-            if not i.deleted and (show_hidden or not i.get('users').get(account_id).get('hide'))
+            # deleted projects have a map node " 'deleted': true' standard projects do not have this property
```
Comment (williamstein) [inline src/smc-webapp/projects.cjsx]: This is still sloppy:

 - There is a single double quote just sitting there in the middle of the line.   Maybe it is supposed to match the single quote after true?   
 - Also, this seems to be two sentences “deleted projects have...” then another sentence “standard project do not...”, but there is not punctuation between them.  Use a period or semicolon.

File: src/smc-webapp/projects.cjsx [hunk 6]
```diff
@@ -525,7 +525,8 @@ class ProjectsStore extends Store
...
+            if (not (i.get('deleted') == true )) and (show_hidden or not i.get('users').get(account_id).get('hide'))
```
Comment (williamstein) [inline src/smc-webapp/projects.cjsx]: The only actual non-comment request I made was related to this line, but you seem not to have seen that.

**Comment by williamstein** [comment]
> -            if not i.deleted and (show_hidden or not i.get('users').get(account_id).get('hide'))
> +            if (not (i.get('deleted') == true )) and (show_hidden or not i.get('users').get(account_id).get('hide'))
> Any reason for the preferred syntax? I wasn't sure what that would return since projects which aren't deleted simply don't have a 'deleted' key/property in their map. If 'deleted' isn't true this is avoided.
> 
Doing 

     if (not (thing == true))

 is completely equivalent in CoffeeScript (and most languages) to doing 

    if not thing

The latter is more readable.   When you’re unsure about what will happen in some case, it’s useful to do some tests or ask.   In this case, if deleted isn’t a key, then `i.get(‘deleted’)` is `undefined`.    Undefined is falsey for the purposes of if/not/etc., not undefined is true.

File: src/smc-webapp/projects.cjsx [hunk 7]
```diff
@@ -525,7 +525,8 @@ class ProjectsStore extends Store
             return 0
         others = []
         for i in v
-            if not i.deleted and (show_hidden or not i.get('users').get(account_id).get('hide'))
+            # Deleted projects have a map node " 'deleted': true ". Standard projects do not have this property.
+            if (not i.get('deleted')) and (show_hidden or not i.get('users').get(account_id).get('hide'))
```
Comment (williamstein) [inline src/smc-webapp/projects.cjsx]: So to summarize, the net relevant change is to replace `i.deleted` by `i.get(‘deleted’)`, which was needed because i is a list of immutable.js objects?

Comment (maarksman) [inline src/smc-webapp/projects.cjsx]: Yes. The variable 'i' is a map element in a list and since 'i.deleted' didn't exist 'not i..deleted' was always true. The deleted property was working in other parts of the site so I figured it probably was not being called properly here.
